server/src/controllers/GrailController.ts

```diff
@@ -86,8 +102,157 @@ export class GrailController {
       dataToSet.runewordData = runewordGrailData;
       return dataToSet;
     });
+
+    this.getMemberParties(address, parties => {
+      parties.forEach(party => {
+        this.saveGrailInPartyDatabase(address, party.address, grailData);
+      });
+    });
   };
 
+  public saveGrailInPartyDatabase = async (
+    grailAddress: string,
+    partyAddress: string,
+    data: any
+  ) => {
+    let partyUserData = GrailController.formatGrailForParty(data);
+    const result = await this.partyCollection.findOneAndUpdate(
+      {
+        address: GrailController.trimAndToLower(partyAddress),
+        "data.users.username": grailAddress
+      },
+      {
+        $set: {
+          "data.users.$.data": partyUserData
+        },
+        $inc: { updateCount: 1 }
+      },
+      { returnOriginal: false }
+    );
+  };
+
+  public static formatGrailForParty = (data: any): any => {
+    let partyGrailData = {
+      uniqueArmor: {
+        missing: 123
+      },
+      uniqueWeapons: {
+        missing: 197
+      },
+      uniqueOther: {
+        missing: 59
+      },
+      sets: {
+        missing: 127
+      },
+      itemScore: 0
```

Comment:
what are these weird default numbers?

Reply:
These are the total items available (add them up = 506).
I had this, because if the user just created their grail and didnt save anything yet from the client, the database data will be empty. So there will not be any items with "wasFound: false" to count for my missingitems methods. This is only used in this one edge case, but this will be removed when changing way db is handled :)